### PR TITLE
Thealmarty/update readme

### PR DESCRIPTION
**Pull request by thealmarty**
- Add dependencies for layout doc generation.
- Add instructions for contributing such that the layout doc will be updated properly.

### CODE REVIEW

File: doc/CONTRIBUTING.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ Contributions are welcome! Please consider the following guidelines.
 - Run `make format` to automatically format your code.
 - Any changes to code style should be submitted as pull requests to https://github.com/cryptiumlabs/ormolu
 
+## Documentation
+
+Add a brief description of any module you create before the module and after pragmas. Start each line of the description with `--`. The first line of the description should be `--|`. E.g., see [this](https://github.com/cryptiumlabs/juvix/blob/15ca9e5e602d24cf09fe87fc059e3e0ee78ad6db/src/Juvix/Encoding/Encoding.hs#L3).
+
```
Comment (mariari) [inline doc/CONTRIBUTING.md]: Seems like a good start, I can probably add details to this soonish.

File: README.md [hunk 2]
```diff
@@ -97,6 +97,8 @@ juvix interactive
 
 [Ormolu](https://github.com/cryptiumlabs/ormolu) required for source formatting.
 
+[Quicklisp](https://www.quicklisp.org/beta/) and [sbcl](http://www.sbcl.org/) required for generation of code layout documentation.
```
Comment (mariari) [inline README.md]: I think something along the lines of what is below would be a bit better.

"required for the automatic generation of documentation in doc/Code" with a link to the folder

Comment (thealmarty) [reply README.md]: Done.

File: doc/CONTRIBUTING.md (hunk 1 above)
Comment (thealmarty) [reply doc/CONTRIBUTING.md]: OK, or you can add 'add suggestions' and I can apply them in this PR too.

**Pull request merged**